pkg/syncer/metadatasyncer.go

```diff
@@ -1041,6 +1043,15 @@ func csiPVDeleted(ctx context.Context, pv *v1.PersistentVolume, metadataSyncer *
 
 		if err := metadataSyncer.volumeManager.DeleteVolume(ctx, volumeHandle, deleteDisk); err != nil {
 			log.Errorf("PVDeleted: Failed to delete disk %s with error %+v", volumeHandle, err)
+			return
```

Comment:
Shouldn't we delete the CR instead of returning from here?

And why do we set deleteDisk to true at line# 1040? Cc @divyenpatel

Reply:
This is for deleting disk upon deleting leftover failed PVs with reclaim policy delete for which PVCs already deleted.

Reply:
Delete reclaim policy will automatically trigger DeleteVolume CSI callbacks, I think metadata syncer should never delete the volume on the datastore. Can we discuss this offline to conclude?

Reply:
sure, this is for `failed` PV for which the controller has given up on those PVs to delete and PVCs already deleted. when we delete such PVs, the controller does not get a call back to delete disks associated with them.

Reply:
That sounds like a bug in Kubernetes to me.

Reply:
>Shouldn't we delete the CR instead of returning from here?

Any comment on this question?

Reply:
When there is an error while deleting volume from CNS, I don't think we should delete the CR. It is better to keep the CR instance when CNS volume is not yet deleted.

Reply:
But the PV is deleted in Kubernetes. I do not see any use in having a CR when the PV is gone.

Reply:
CR is a mapping for both PV object and CNS object. If FCD is still not deleted we should not delete the mapping resource(CR). I am only thinking in terms of correctness.
@divyenpatel  Any comments here ?

Reply:
> That sounds like a bug in Kubernetes to me.

PV gets `Failed` state if k8s fails to be delete PV after it is released from claim. and when user delete such PV manually,  associated disk remain on the datastore. so to clean up disk associated with this PV we have added this logic.

Another use case is, if user creates PV statically with reclaim policy set to delete. and then deletes this PV without creating PVC to bind with this PV, we should delete the associated disk upon deleting PV using metadata syncer. 
Here since the claim is not associated with PV yet, the controller does not help delete disk.

we have this logic since very first version of our CSI driver.

Reply:
> But the PV is deleted in Kubernetes. I do not see any use in having a CR when the PV is gone.

@SandeepPissay I discussed with Divyen and we agree with your comment and removed the return statement, so that voluemigration CR gets deleted when PV is gone in k8s